solver/src/test/java/org/evomaster/solver/Z3DockerExecutorTest.java

```diff
@@ -78,44 +78,63 @@ public void dynamicFile() throws IOException {
-     * And the response is:
-     * ((users1 (id-name-age-points 3 "agus" 31 7)))
+     * Test solving a model with composed types
      */
     @Test
     public void composedTypes() {
         Optional<Map<String, SMTLibValue>> response = executor.solveFromFile("composed_types.smt");
 
-        assertTrue(response.isPresent());
+        assertTrue(response.isPresent(), "Response should be present for composed_types.smt");
+
+        assertTrue(response.get().containsKey("users1"), "Response should contain users1");
+        Map<String, SMTLibValue> users1Expected = ImmutableMap.of(
+                "ID", new IntValue(3),
+                "NAME", new StringValue("agus"),
+                "AGE", new IntValue(31),
+                "POINTS", new IntValue(7)
+        );
+        assertEquals(new StructValue(users1Expected), response.get().get("users1"), "The value for users1 is incorrect");
+
+        assertTrue(response.get().containsKey("users2"), "Response should contain users2");
+        Map<String, SMTLibValue> users2Expected = ImmutableMap.of(
+                "ID", new IntValue(3),
+                "NAME", new StringValue("agus"),
+                "AGE", new IntValue(31),
+                "POINTS", new IntValue(7)
+        );
+        assertEquals(new StructValue(users2Expected), response.get().get("users2"), "The value for users2 is incorrect");
+    }
+
+    /**
+     * Test solving with an invalid file to ensure proper error handling
+     */
+    @Test
+    public void invalidFile() {
+        try {
+            executor.solveFromFile("invalid.smt");
+        } catch (Exception e) {
+            assertEquals("No result after solving file ", e.getMessage(), "Exception message should be 'error: file is empty'");
+        }
+    }
 
-        assertTrue(response.get().containsKey("users1"));
-        Map<String, SMTLibValue> users1Expected = new java.util.HashMap<>();
-        users1Expected.put("ID", new IntValue(3));
-        users1Expected.put("NAME", new StringValue("agus"));
-        users1Expected.put("AGE", new IntValue(31));
-        users1Expected.put("POINTS", new IntValue(7));
-        assertEquals(new StructValue(users1Expected), response.get().get("users1"));
-
-        assertTrue(response.get().containsKey("users2"));
-        Map<String, SMTLibValue> users2Expected = new java.util.HashMap<>();
-        users2Expected.put("ID", new IntValue(3));
-        users2Expected.put("NAME", new StringValue("agus"));
-        users2Expected.put("AGE", new IntValue(31));
-        users2Expected.put("POINTS", new IntValue(7));
-        assertEquals(new StructValue(users2Expected), response.get().get("users2"));
+    /**
+     * Test handling an empty file
+     */
+    @Test
+    public void emptyFile() {
+        try {
+            executor.solveFromFile("empty.smt");
+        } catch (Exception e) {
+            assertEquals("No result after solving file ", e.getMessage(), "Exception message should be 'error: file is empty'");
```

Comment:
brittle check. if we change the error message, then this test fails. 
if really need to check this, the error message could be put in a static final variable, and not string duplicated here